server/services/voip/connector/asterisk/IConnection.ts

```diff
@@ -0,0 +1,43 @@
+
+/**
+ * Delegate interface for Asterisk management connection.
+ * @remarks
+ * This interface will be implemented by all those classes which are
+ * interested in connecting to asterisk via different ways such as AMI, ARI and AGI
+ *
+ * */
+export interface IConnection {
+	/**
+	 * Called for conneting to the server
+	 * @remarks
+	 * Callback for handling incoming call
+	 */
+	connect?(
+		connectionIpOrHostname: string,
+		connectionPort: string,
+		userName: string,
+		password: string,
+	): void;
+	/**
+	 * Called for executing the command to the connection
+	 * @remarks
+	 */
+	executeCommand?(action: object, actionResultCallback: any): void;
```

Comment:
```suggestion
	executeCommand(action: object, actionResultCallback: any): void;
```

Reply:
I think we mostly won't need an extra question mark at the end of all function name. Please correct me if I'm missing something here

Reply:
Makes sense. Removing optional interface methods wherever applicable.